deeplay/components/transformer/enc.py

```diff
@@ -0,0 +1,136 @@
+from typing import List, Optional, Literal, Any, Sequence, Type, overload, Union
+
+from .. import DeeplayModule, Layer, LayerList
+from . import LayerSkipNormalization, MultiheadSelfAttention
+
+from deeplay.blocks.sequential import SequentialBlock
+from deeplay import Sequential
+
+import torch
+import torch.nn as nn
+
+import itertools
+from functools import reduce
+
+
+class Add(DeeplayModule):
+    def forward(self, *x):
+        """
+        Add the input tensors element-wise.
+        """
+        x = itertools.chain(*x) if isinstance(x[0], tuple) else x
+        return reduce(lambda a, b: torch.add(a, b), x)
+
+
+class TransformerEncoderLayer(DeeplayModule):
```

Comment:
Should contain some documentation in comments, similarly to the the other components

Reply:
Done!